[Doc] Update the tutorial to use Vite instead of create-react-app (#8351)
marmelab/react-admin: docs/Tutorial.md

## Setting Up

React-admin uses React. We'll use [vite.js](https://vitejs.dev/) to create an empty React app, and install the `react-admin` package:

```sh
yarn create vite test-admin --template react-ts
cd test-admin/
yarn add react-admin ra-data-json-server
yarn dev
```

You should be up and running with an empty React application on port 5173.

> **Tip**: Although this tutorial uses a TypeScript template, you can use react-admin with JavaScript if you prefer. Also, you can use [create-react-app](./CreateReactApp.md), [Next.js](./NextJs.md), [Remix](./Remix.md), or any other React framework to create your admin app. React-admin is framework agnostic.

Reviewer: ```suggestion
**Tip**: Although this tutorial uses a TypeScript template, you can use react-admin with JavaScript if you prefer. Also, you can use [create-react-app](./CreateReactApp.md), [Next.js](./NextJs.md), [Remix](./Remix.md), or any other React framework to create your admin app. React-admin is framework-agnostic.
```

## Using an API As Data Source

> React-admin runs in the browser, and fetches data from an APIs.

Reviewer 2: ```suggestion
React-admin runs in the browser, and fetches data from an API.
```

> We'll be using [JSONPlaceholder](https://jsonplaceholder.typicode.com/), a fake REST API designed for testing and prototyping, as the datasource for the application. Here is what it looks like:

Reviewer: ```suggestion
We'll be using [JSONPlaceholder](https://jsonplaceholder.typicode.com/), a fake REST API designed for testing and prototyping, as the data source for the application. Here is what it looks like:
```

{% raw %}
```jsx
// in src/MyUrlField.tsx
import { useRecordContext } from "react-admin";
import { Link } from "@mui/material";
import LaunchIcon from "@mui/icons-material/Launch";

> const MyUrlField = ({ source }: { source: string }) => {

Reviewer: ```suggestion
const MyUrlField = ({ source }) => {
```

React-admin can use Input components to create a multi-criteria search engine in the list view. Pass an array of such Input components to the List `filters` prop to enable filtering:

```jsx
// in src/posts.js

> import { ReferenceInput, TextInput, List } from 'react-admin';

Reviewer 2: why remove this import?

Author: Because the import doesn't change from the previous step, and it doesn't look like that in the previous step. As is, the new import is misleading.

The `authProvider` must expose 5 methods, each returning a `Promise`:

```jsx
// in src/authProvider.ts

> export const authProvider = {

Reviewer: Improve indentation

Author: It's the same indentation as the other snippets

Reviewer: Ok then

  // called when the user attempts to log in

>   login: ({ username }: { username: string }) => {

Reviewer: ```suggestion
  login: ({ username }) => {
```

    localStorage.setItem("username", username);
    // accept all username/password combinations
    return Promise.resolve();
  },
  // called when the user clicks on the logout button
  logout: () => {
    localStorage.removeItem("username");
    return Promise.resolve();
  },
  // called when the API returns an error

>   checkError: ({ status }: { status: number }) => {

Reviewer: ```suggestion
  checkError: ({ status }) => {
```

The code for a Data Provider for the `my.api.url` API is as follows:

```js

> import { fetchUtils } from "react-admin";

Reviewer 2: Suggestion: change also the line below:

```diff
- import { stringify } from 'query-string';
+ import { stringify } from "query-string";
```

import { stringify } from 'query-string';

const apiUrl = 'https://my.api.com/';
const httpClient = fetchUtils.fetchJson;

> export const dataProvider= {

Reviewer 2: Should we change this to TS as well? Like declaring `dataProvider` as being of type `DataProvider`?

Author: No, all the snippets in the tutorial are in JS.
